modules/transactionpool/transactionpool.go

```diff
@@ -151,3 +153,80 @@ func (tp *TransactionPool) TransactionList() []types.Transaction {
 	}
 	return txns
 }
+
+// Transaction returns the transaction with the provided txid, its parents, and
+// a bool indicating if it exists in the transaction pool.
+func (tp *TransactionPool) Transaction(id types.TransactionID) (types.Transaction, []types.Transaction, bool) {
+	tp.mu.Lock()
+	defer tp.mu.Unlock()
+
+	// find the transaction
+	exists := false
+	var txn types.Transaction
+	for _, tSet := range tp.transactionSets {
+		for _, t := range tSet {
+			if t.ID() == id {
+				txn = t
+				exists = true
+			}
+		}
+	}
+
+	// collect parent IDs
```

Comment:
not good enough, also not needed. You don't just want the parents, you want all the grandparents, etc. as well. But you already have all those, they are cluseterd in to transactionSets. So all you really need to do with a transaction is use the transaction set as the set of parents. The transaction set is ordered correctly, so if the txn is not the last item of the set, you can just use the elements prior.

There's an edge case where you may end up with unneeded parents. So to be completely correct you have to loop through the parents in reverse order and check if they are in fact necessary. And if so, you expand the set of relevant outputs, and continue the loop. Overall, it should be pretty fast to do that, it should be a linear time operation.